### PR TITLE
6.1.1 I95-51052 Release Note

### PR DESCRIPTION
**Pull request by Chr1st0ph3rTurn3r**

### CODE REVIEW

File: docs/release_notes_128t_6.1.md [hunk 1]
```diff
@@ -49,6 +49,18 @@ This issue will be corrected in an upcoming release.
 The Juniper SSR team does not publicly disclose known or resolved CVEs in our public documentation but instead utilizes our internal bug tracking IDs. Customers can obtain the actual CVE numbers by contacting Juniper Support.
 :::
 
+## Release 6.1.1-3
+
+**Release Date:** April 28, 2023
+
+### Resolved Issues
+
+- **I95-51052 `session-scaling` regression generating high number of `wrong-thread` error messages:** The following regression has been resolved: If `outbound-only` or `nat-traversal mode` is used between SSR routers, and a NAT between the routers causes an address and/or port change for existing TCP or UDP sessions between them, the SSR did not automatically recover those sessions if they were idle or had one-way traffic.
```
Comment (MichaelBaj) [inline docs/release_notes_128t_6.1.md:58]: Change to:

**I95-51052 NAT changes in network cause sessions to fail:** If `session-scaling` is set to enabled with `outbound-only` or `nat-traversal mode` is used between SSR routers, and a NAT between the routers causes an address and/or port change for existing TCP or UDP sessions between them, the SSR will not automatically recover those sessions if they were idle or had one-way traffic.

The impacted sessions will time out when all packets for the failed sessions stopped. New sessions are not impacted, nor are any pre-existing sessions with regular bi-directional traffic.

**Review by MichaelBaj — CHANGES_REQUESTED** [review]

File: docs/release_notes_128t_6.1.md (hunk 1 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/release_notes_128t_6.1.md]: Updated

**Review by MichaelBaj — DISMISSED** [review]

File: docs/release_notes_128t_6.1.md [hunk 2]
```diff
@@ -49,6 +49,16 @@ This issue will be corrected in an upcoming release.
 The Juniper SSR team does not publicly disclose known or resolved CVEs in our public documentation but instead utilizes our internal bug tracking IDs. Customers can obtain the actual CVE numbers by contacting Juniper Support.
 :::
 
+## Release 6.1.1-3
+
+**Release Date:** April 28, 2023
+
+### Resolved Issues
+
+- **I95-51052 NAT changes in network cause sessions to fail:** If `session-scaling` is set to `enabled` with `outbound-only` or `nat-traversal` mode used between SSR routers, and a NAT between the routers causes an address and/or port change for existing TCP or UDP sessions between them, the SSR will not automatically recover those sessions if they were idle or had one-way traffic.
```
[the remark this answers is not in the document]
Comment (hadrielk) [reply docs/release_notes_128t_6.1.md:58]: Lol, **you** were the one who told me to use that term!

Comment (hadrielk) [inline docs/release_notes_128t_6.1.md:58]: @Chr1st0ph3rTurn3r in the end, after more analysis internally, I think this wording is a bit too extreme and broad. It takes a specific traffic pattern to hit this issue, an it's not a common one. (I mean, it's common enough that Commerzbank hit it, but it's not like every session has the problem)

**Review by hadrielk — CHANGES_REQUESTED** [review]

**Review by ayungelson — APPROVED** [review]

**Comment by hadrielk** [comment]
FYI: we're also going to need to release-note this PR: https://github.com/Juniper-SSN/ssr/pull/6640

That's being merged in as well. I added a potential red-note to the PR description, but could use word-smithing.

I'm not sure howe we publicly word fixes for race-condition crashes.

File: docs/release_notes_128t_6.1.md [hunk 3]
```diff
@@ -58,6 +58,8 @@ The Juniper SSR team does not publicly disclose known or resolved CVEs in our pu
 - **I95-51052 NAT changes in network may cause sessions to fail:** If `session-scaling` is set to `enabled` with `outbound-only`, and a NAT between the routers causes an address and/or port change for existing TCP or UDP sessions between them, the SSR may not automatically recover idle or one-way traffic sessions.
 
 The impacted sessions will time out when all packets for the failed sessions stop. New sessions are not impacted, nor are any pre-existing sessions with regular bi-directional traffic.
+------
+- **I95-51093 95-51093 Race condition in `session-scaling` can cause crash/restart:** Resolved an issue when `session-scaling` is set to `enabled` together with `outbound-only`, a race condition can cause a crash and restart of the `highway` process in the SSR.
```
Comment (MichaelBaj) [inline docs/release_notes_128t_6.1.md:62]: Missing 'I' in I95 issue ID.

**Review by MichaelBaj — APPROVED** [review]

**Review by ayungelson — APPROVED** [review]

**Review by hadrielk — APPROVED** [review]

**Review by MichaelBaj — DISMISSED** [review]

File: docs/release_notes_128t_6.1.md [hunk 4]
```diff
@@ -49,14 +49,15 @@ This issue will be corrected in an upcoming release.
...
-## Release 6.1.1-5
+## Release 6.1.1-6
 
 **Release Date:** April 28, 2023
 
 ### Resolved Issues
 
-- **I95-51052 NAT changes in network may cause sessions to fail:** If `session-scaling` is set to `enabled` with `outbound-only`, and a NAT between the routers causes an address and/or port change for existing TCP or UDP sessions between them, the SSR may not automatically recover idle or one-way traffic sessions.
-
+- **I95-48965, I95-50070, I95-50086 Highway Crashes:** Resolved an issue where a routing change that affects the `forwarding-table` can incur a race condition with sessions completing and being removed, which could lead to a highway crash and restart.
```
Comment (MichaelBaj) [inline docs/release_notes_128t_6.1.md:58]: [I95-50086](https://128technology.atlassian.net/browse/I95-50086) is unrelated from other two items and is NOT in 6.1.1. It is in 6.0.10.

I95-50070 is a duplicate of I95-48965 and should be removed from the release notes.

Headline should be changed to 
**I95-48965 Race condition with routing updates inducing crash in highway process:** ...

**Review by MichaelBaj — CHANGES_REQUESTED** [review]

File: docs/release_notes_128t_6.1.md (hunk 4 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/release_notes_128t_6.1.md:58]: So two things with this. One is that I got the wrong number - it should be 51086, which I'll fix. Two, Jeff asked me to document all three in the release notes. I don't know if CHS and Plusnet are the same customer - but if they are two different customers, and this is a high visibility issue, should they both be in there? And I'll change the Issue name.

Comment (MichaelBaj) [reply docs/release_notes_128t_6.1.md:58]: These are two different customers. Therefore you _can_ add both issue IDs for that purpose if you feel that it is valuable.

**Review by MichaelBaj — DISMISSED** [review]

File: docs/release_notes_128t_6.1.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeffmart-jnpr) [reply docs/release_notes_128t_6.1.md]: nat-traveral mode section was removed to resolve this comment

**Review by ayungelson — APPROVED** [review]

File: docs/release_notes_128t_6.1.md (hunk 1 above)
Comment (jeffmart-jnpr) [reply docs/release_notes_128t_6.1.md]: confirmed + changes requested below.

File: docs/release_notes_128t_6.1.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeffmart-jnpr) [reply docs/release_notes_128t_6.1.md]: confirmed, resolving ...

File: docs/release_notes_128t_6.1.md (hunk 4 above)
Comment (jeffmart-jnpr) [reply docs/release_notes_128t_6.1.md]: I do, resolving this section :)

**Review by jeffmart-jnpr — APPROVED** [review]

**Review by MichaelBaj — APPROVED** [review]

**Pull request merged**